analysis.go

```diff
@@ -62,7 +63,7 @@ func NewAnalyzer(c *Config) (*analysis.Analyzer, error) {
 					defer wg.Done()
 
 					for file := range jobCh {
-						filename := pass.Fset.PositionFor(file.Pos(), false).Filename
+						filename := pass.Fset.PositionFor(file.Pos(), true).Filename
 						if !strings.HasSuffix(filename, ".go") {
 							continue
 						}
```

Comment:
```suggestion
						filename := pass.Fset.PositionFor(file.Pos(), true).Filename
						if !strings.HasSuffix(filename, ".go") {
							continue
						}

						// if the file is related to cgo the filename of the unadjusted position is a not a '.go' file.
						if !strings.HasSuffix(pass.Fset.PositionFor(file.Pos(), false).Filename, ".go") {
							continue
						}
```

Reply:
because you cannot process it: the file content inside the pass is not the real file content.

Reply:
I've tested the change locally, and it's working wrong.  Cgo files with original content were just ignored.

Note: the original line will allow processing files with real content, i.e., files with go ext. It's similar with https://github.com/golangci/golangci-lint/blob/main/pkg/goanalysis/position.go#L11 (in case when it's returning true)

Reply:
The goal is to ignore cgo files, so yes it ignores cgo files.

> the original line will allow processing files with real content

It doesn't work like that.

Reply:
finally disabled cgo checks by default for now